### PR TITLE
Implement new defintions suggested by @jadephilipoom

### PR DESCRIPTION
**Pull request by aferr**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: experimental/ni_coq/vfiles/PossibilisticNI.v
Comment (aferr) [inline experimental/ni_coq/vfiles/PossibilisticNI.v:500]: @jadephilipoom Can you take a look at this part?

File: experimental/ni_coq/vfiles/PossibilisticNI.v
Comment (aferr) [inline experimental/ni_coq/vfiles/PossibilisticNI.v:517]: @jadephilipoom (Alternatively see if this lemma is really needed).

File: experimental/ni_coq/vfiles/NIUtilTheorems.v
Comment (aferr) [inline experimental/ni_coq/vfiles/NIUtilTheorems.v:201]: @jadephilipoom and actually also can you take a look at this to make sure FE is not needed here either?

File: experimental/ni_coq/vfiles/PossibilisticNI.v
Comment (jadephilipoom) [reply experimental/ni_coq/vfiles/PossibilisticNI.v:500]: I think it can't, but I might have an idea for how to avoid needing it. Testing it out and will update you soon!

Comment (jadephilipoom) [reply experimental/ni_coq/vfiles/PossibilisticNI.v:500]: Okay, it does work! Pushed the code to my fork [here](https://github.com/jadephilipoom/oak/tree/new_state_loweq). ([diff](https://github.com/jadephilipoom/oak/commit/2cb427d2c812364cd50c04d7981f368c926f6621))

I've replaced this lemma with:
```coq
￼Lemma step_system_ev_projection ell s1 s2 s3 e :
￼  state_low_eq ell s1 s2 ->
￼  step_system_ev (state_low_proj ell s2) s3 e ->
￼  (exists s4, state_low_eq ell s3 s4
￼         /\ step_system_ev (state_low_proj ell s1) s4 e).
```

This means that, without proving Leibniz equality between the `s1` and `s2` projections, we can still manipulate a`step_system_ev` hypothesis to get a `step_system_ev` hypothesis for the projection of an equivalent state.

File: experimental/ni_coq/vfiles/NIUtilTheorems.v
Comment (jadephilipoom) [reply experimental/ni_coq/vfiles/NIUtilTheorems.v:201]: Adding `congruence` at the end solved this lemma!

File: experimental/ni_coq/vfiles/NIUtilTheorems.v
Comment (jadephilipoom) [inline experimental/ni_coq/vfiles/NIUtilTheorems.v:222]: I'm a little surprised that this proof broke. Would swapping `congruence` for `auto` solve it?

**Review by jadephilipoom — APPROVED** [review]
LGTM! If you want to use the changes I linked to, I can either push them to this branch or create a separate PR once this is merged.

File: experimental/ni_coq/vfiles/NIUtilTheorems.v
Comment (aferr) [reply experimental/ni_coq/vfiles/NIUtilTheorems.v:222]: It looks like that theorem isn't actually used, so it can also just be deleted. If you push your change to this branch would you mind either checking if that easy change proves it, or just deleting the theorem?

**Comment by aferr** [comment]
Awesome! It's nice that we can skip adding an axiom. Thanks for working on this. If you push your change to this branch we have one fewer PR to do, so that seems slightly easier. (Though I don't have a strong preference if you'd rather do it the other way).

File: experimental/ni_coq/vfiles/NIUtilTheorems.v
Comment (aferr) [reply experimental/ni_coq/vfiles/NIUtilTheorems.v:222]: (And if it turns out to be more work than either of those trivial things don't worry about it).

Comment (jadephilipoom) [reply experimental/ni_coq/vfiles/NIUtilTheorems.v]: Swapping `congruence` for `eauto` did work, so I've pushed that.

**Pull request merged**
